WordPress/gutenberg
# Removing Reakit `Composite` implementation

## What?

Removes the 'unstable' Reakit implementation of the `Composite` component group.

## Why?

We need to remove Reakit from the codebase (#53278).

## How?

* Points default export to `./legacy` implementation
* Changes 'legacy' Storybook title to "Composite"
* Removes deprecation messages from _legacy_ (to be put back when _current_ is available)
* Removes _unstable_ Reakit implementation

## Testing Instructions

Open Storybook, and confirm that there are still two `Composite` entries ("Composite" and "Composite (V2)").

![Storybook sidebar items, with "Composite (V2)" and "Composite" highlighted.](https://github.com/WordPress/gutenberg/assets/159848/06a53f10-719c-456f-bb90-5120abc7cbba)

> [!NOTE]
> There are no consumers of the Reakit implementation left in Gutenberg, so we have to trust that test coverage is sufficient that any switch is transparent for consumers.
